public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -106,17 +106,29 @@ export class PrometheusDatasource {
     });
 
     return this.$q.all(allQueryPromise).then(responseList => {
-      var result = [];
+      let result = [];
 
       _.each(responseList, (response, index) => {
         if (response.status === 'error') {
           throw response.error;
         }
 
+        let prometheusResult = response.data.data.result;
+
         if (activeTargets[index].format === 'table') {
-          result.push(self.transformMetricDataToTable(response.data.data.result, responseList.length, index));
+          result.push(self.transformMetricDataToTable(prometheusResult, responseList.length, index));
```

Comment:
This class is now 500 rows. I think its time we start breaking it down into smaller pieces. How about a class for transforming and parsing the response? If I remember correctly it was a little bit messy to extract it but I think it would be good for maintainability.

Reply:
Yes, that was in your branch, so I can split this file.